runtimes/server-interface/workspace.ts

```diff
@@ -27,4 +30,5 @@ export type Workspace = {
         isFile: (path: string) => Promise<boolean>
         remove: (dir: string) => Promise<void>
     }
+    onCursorStateUpdate: (handler: NotificationHandler<CursorStateUpdateParams>) => void
```

Comment:
nit: `onCursorStateChange` fits a bit better for the name

Reply:
If we included cursor position into chat & quick actions request, it would have been much less chatty.

Reply:
Yes, for now those two would be the only two places where we add the cursorstate param. But it can cause issues as the interfaces grow, we would need to add the cursor state as an optional parameter into many more requests